src/lexer/scan.ts

```diff
@@ -233,9 +233,13 @@ export function scanSingleToken(parser: ParserState, context: Context, state: Le
             if (ch === Chars.Exclamation) {
               // Treat HTML begin-comment as comment-till-end-of-line.
               if (
-                parser.source.charCodeAt(parser.index + 2) === Chars.Hyphen &&
```

Comment:
This code is an optimization tweak. Checking for the 2nd 'Chars.Hypen' first avoids extra check for the first  'Chars.Hyphen' if 2nd doesn't exist. Kind of micro-optimization.

Reply:
Ooh I like it, but I was worried from the edge case that `parser.index + 1 == EOF` and there's no `parser.index + 2`. I guess NaN will return which is kind of OK, but I wanted to avoid that. Should we maybe check that `pos` doesn't exceed `parser.end + 2`?

Reply:
Better yes to check for `parser.end + 2`.